pkg/volume/volume.go

```diff
@@ -28,6 +28,28 @@ import (
 type Volume interface {
 	// GetPath returns the directory path the volume is mounted to.
 	GetPath() string
+
+	// VolumeMetricsProvider embeds methods for exposing Volume metrics (e.g. capacity).
+	VolumeMetricsProvider
+}
+
+// VolumeMetricsProvider exposes metrics (e.g. capacity) related to a Volume.
+type VolumeMetricsProvider interface {
+	// GetCapacityMetrics returns the CapacityMetrics for this Volume.  Maybe expensive for some implementations.
+	GetCapacityMetrics() (*CapacityMetrics, error)
+}
+
+// CapacityMetrics represents available and consumed capacity for the Volume.
+type CapacityMetrics struct {
+	// VolumeBytesUsed represents the total bytes used under the directory path that the Volume is mounted.
+	// Note: For block devices this maybe more than the total size of the files.  See `man du` for details.
+	VolumeBytesUsed uint64
```

Comment:
Assuming that we will follow similar naming conventions for the kubelet metrics APIs, I prefer not exposing filesystem specific concepts.
How about exposing 
1. `Capacity` - The overall capacity of a volume. In the case of shared storage volumes, this will be size of the underlying storage.
2. `Usage` - Usage of a volume. 
3. `Free` - Storage space available for a volume. In the case of shared storage volumes, `Free` will not be equal to `Capacity` - `Usage`.